### PR TITLE
1314902: ENT-555 use dmi.system.uuid to reconcile hypervisors

### PR DESCRIPTION
**Pull request by vritant**

### CODE REVIEW

File: server/src/main/java/org/candlepin/model/ConsumerCurator.java [hunk 1]
```diff
@@ -699,6 +700,71 @@ public VirtConsumerMap getHostConsumersMap(Owner owner, Iterable<String> hypervi
         return hypervisorMap;
     }
 
+    /**
```
[the remark this answers is not in the document]
Comment (vritant) [reply server/src/main/java/org/candlepin/model/ConsumerCurator.java]: @mstead async version has the fact we are using here, sync version does not.
only a recent version of virt-who would provide the fact in the report, and all recent versions should talk to the async end point anyway.

**Comment by vritant** [comment]
@mstead done

File: server/src/main/java/org/candlepin/model/ConsumerCurator.java [hunk 2]
```diff
@@ -699,6 +700,71 @@ public VirtConsumerMap getHostConsumersMap(Owner owner, Iterable<String> hypervi
...
+
+        Map<String, HypervisorId> systemUuidHypervisorMap = new HashMap<>();
+        List<String> remainingHypervisorIds = new LinkedList<>();
+        for (Consumer consumer : hypervisorList.getHypervisors()) {
+            if (consumer.hasFact("dmi.system.uuid")) {
+                systemUuidHypervisorMap.put(consumer.getFact("dmi.system.uuid"), consumer.getHypervisorId());
+            }
+            remainingHypervisorIds.add(consumer.getHypervisorId().getHypervisorId());
+        }
+        List<String> consumerIds = new LinkedList<>();
```
Comment (mstead) [inline server/src/main/java/org/candlepin/model/ConsumerCurator.java]: This should be moved into the block it is used in.

**Comment by wottop** [comment]
Code looks good, but.

If a consumer with the system.uuid comes in first, the hostname in CP is set to the system.uuid.
What happens when the hypervisor report comes in with a 'machine name' hostname?
I don't think you changed the behavior from what I had, but we should reconcile it. The Satellite folks will expect the hostname in the hypervisor report as soon as it is reported.

**Comment by vritant** [comment]
@wottop added the logic, and a new unit test to test it too

**Comment by mstead** [comment]
Ooops.. Clicked the close button by mistake.

**Comment by vritant** [comment]
@wottop @mstead any objection to merge this PR?

**Comment by wottop** [comment]
ack

**Pull request merged**